include/wil/registry.h

```diff
@@ -17,16 +17,3139 @@
+        {
+            const reg_view_details::reg_view_nothrow regview{ key };
+            return regview.open_key(path, hkey, access);
+        }
+
+        /**
+         * \brief Creates a new HKEY to the specified path
+         *        see RegCreateKeyExW
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param path A string to append to the HKEY to attempt to open
+         *        can be nullptr if not needed
+         * \param[out] hkey A pointer to an HKEY parameter to receive the opened HKEY
+         * \param access The requested access desired for the opened key
+         * \return HRESULT error code indicating success or failure
+         *         does not throw C++ exceptions
+         */
+        inline HRESULT create_key_nothrow(HKEY key, PCWSTR path, _Out_ HKEY* hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) WI_NOEXCEPT
+        {
+            const reg_view_details::reg_view_nothrow regview{ key };
+            return regview.create_key(path, hkey, access);
+        }
+
+#if defined(__SDDL_H__)
+        /**
+         * \brief Creates a new HKEY to the specified path
+         *        see RegCreateKeyExW
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param path A string to append to the HKEY to attempt to open
+         *        can be nullptr if not needed
+         * \param security_descriptor A string-format security descriptor specifying the security descriptor for the new HKEY
+         *        see ConvertStringSecurityDescriptorToSecurityDescriptorW
+         * \param[out] hkey A pointer to an HKEY parameter to receive the opened HKEY
+         * \param access The requested access desired for the opened key
+         * \return HRESULT error code indicating success or failure
+         *         does not throw C++ exceptions
+         *
+         * \remark Because security functions and types are referenced, the <sddl.h> header must be previously included
+         */
+        inline HRESULT create_key_nothrow(HKEY key, PCWSTR path, _In_opt_ PCWSTR security_descriptor, _Out_ HKEY* hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) WI_NOEXCEPT
+        {
+            const reg_view_details::reg_view_nothrow regview{ key };
+            return regview.create_key(path, security_descriptor, hkey, access);
+        }
+#endif // #if defined(__SDDL_H__)
+
+
+#if defined(WIL_ENABLE_EXCEPTIONS)
+        /**
+         * \brief Queries for number of sub-keys
+         * \param key The HKEY to query for number of sub-keys
+         * \return The queried number of sub-keys if succeeded
+         * \exception std::exception will be thrown on failure (including wil::ResultException)
+         */
+        inline size_t get_child_key_count(HKEY key)

```

Comment:
Should we return the actual type used (`DWORD`)? Since it doesn't really support up to `size_t`.

Reply:
yep. should change this.

Reply:
We should also remove the `uint32_t` nothrow overloads in that case. Let callers decide how to cast the number.

Reply:
To harmonize with the rest of the API, no need to remove the `uint32_t` overloads.